src/main/resources/assets/computercraft/lua/rom/programs/http/gist.lua

```diff
@@ -0,0 +1,335 @@
+--- gist.lua - Gist client for ComputerCraft
+-- Made by JackMacWindows for CraftOS-PC and CC: Tweaked
+--
+-- @module gist
+
+local expect = require and require("cc.expect").expect or dofile("/rom/modules/main/cc/expect.lua").expect
+
+-- Internal functions
+
+local function getGistFile(data)
+    if not data.truncated then return data.content else
+        local handle = http.get(data.raw_url)
+        if not handle then error("Could not connect to api.github.com.") end
+        if handle.getResponseCode() ~= 200 then
+            handle.close()
+            error("Failed to download file data.")
+        end
+        local d = handle.readAll()
+        handle.close()
+        return d
+    end
+end
+
+local function setTextColor(c) if term.isColor() then term.setTextColor(c) elseif c == colors.white or c == colors.yellow then term.setTextColor(colors.white) else term.setTextColor(colors.lightGray) end end
+
+local function requestAuth(headers, interactive)
+    if settings.get("gist.id") ~= nil then headers.Authorization = "token " .. settings.get("gist.id") elseif interactive then
+        setTextColor(colors.yellow)
+        write("You need to add a Personal Access Token (PAK) to upload Gists. Follow the instructions at ")
+        setTextColor(colors.blue)
+        write("https://tinyurl.com/GitHubPAK")
+        setTextColor(colors.yellow)
+        write(" to generate one. Make sure to check the '")
+        setTextColor(colors.blue)
+        write("gist")
+        setTextColor(colors.yellow)
+        print("' checkbox on step 7 (under 'Select scopes'). Once done, paste it here.")
+        setTextColor(colors.lime)
+        write("PAK: ")
+        setTextColor(colors.white)
+        local pak = read()
+        if pak == nil or pak == "" then error("Invalid PAK, please try again.") end
+        settings.set("gist.id", pak)
```

Comment:
Do we want to save the settings afterwards?

Reply:
I wasn't sure where to save it at first, but I updated it to save to `/.settings`.